### ISSUE TITLE
visual results

### ISSUE THREAD

**Issue by GeGehub** [bcmi/Object-Shadow-Generation-Dataset-DESOBAv2#8]
Can you provide your visual results on the test set? I tried to infer your set but found very bad results. Thanks

**Comment by ustcnewly** [comment]
You can use our released model to get the test results. For small objects or objects with simple shapes, the generated shadows are usually acceptable. For large objects or objects with complex shapes, the results are often unsatisfactory, but you can sample multiple times and get a relatively good result.

You can also try the shadow generation function in our libcom toolbox https://github.com/bcmi/libcom.

**Comment by GeGehub** [comment]
![image](https://github.com/bcmi/Object-Shadow-Generation-Dataset-DESOBAv2/assets/118744986/ed2df322-9f0a-4ca1-959a-14d140945b68)
Thanks for your reply. The above figure is the results I tested. I just wondering if these results are reasonable? In addition, I saw that in your test set, most images contain only one object. In this case, How does your model learn the intensity through the intensity encoder?

**Comment by Narumi-Maria** [comment]
> By the way, as your dataset contains both 256 resolution and hr, what is the used resolution for training and testing?

To align with previous work, we conduct testing and training at a resolution of 256.

**Comment by GeGehub** [comment]
![image](https://github.com/bcmi/Object-Shadow-Generation-Dataset-DESOBAv2/assets/118744986/20108c7e-3208-485d-8ce0-d7d11fb64493)
I tried to re-train your model without modifying any code, but the model seems barely converged. The test result is shown in above.

**Comment by Narumi-Maria** [comment]
> ![image](https://private-user-images.githubusercontent.com/118744986/339955810-20108c7e-3208-485d-8ce0-d7d11fb64493.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg0MzU0ODQsIm5iZiI6MTcxODQzNTE4NCwicGF0aCI6Ii8xMTg3NDQ5ODYvMzM5OTU1ODEwLTIwMTA4YzdlLTMyMDgtNDg1ZC04Y2UwLWQ3ZDExZmI2NDQ5My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxNVQwNzA2MjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZDI5YzZmZGJmNTBmMzUzOGU1MDc4OWU5MzgwMmIzNWJkYTc2M2U0YWQ1ZTZhZGJmY2YyOTRmZjUwOGMzZmZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.1z4KfM7JLcJThZQ8dx7uF6slxfJ8YkA_k5LEsPr6Y3s) I tried to re-train your model without modifying any code, but the model seems barely converged. The test result is shown in above.

you may need to load pre-trained weights for controllnet

**Comment by GeGehub** [comment]
1. Could you please add this code?
2. Which weights should I load? Only the Controlnet or the whole Diffusion UNet?

**Comment by Narumi-Maria** [comment]
model = create_model('./models/cldm_v15.yaml').cpu()
model_weight = load_state_dict(resume_path, location='cpu', strict=False)
rgb_weight = model_weight['control_model.input_hint_block.0.weight']
gray_weight = 0.2989 * rgb_weight[:, 0:1, :, :] + 0.5870 * rgb_weight[:, 1:2,: ,:] + 0.1140 * rgb_weight[:, 2:3,: ,:]
model_weight['control_model.input_hint_block.0.weight'] = torch.cat([rgb_weight, gray_weight], dim=1)

You can download the file "control_sd15_ini.ckpt" from the URL "https://huggingface.co/runwayml/stable-diffusion-v1-5/tree/main" and use the local path where you store "control_sd15_ini.ckpt" as the resume_path.
